datadog_lambda/tracing.py

```diff
@@ -1320,6 +1314,10 @@ def create_inferred_span_from_eventbridge_event(event, context):
     if span:
         span.set_tags(tags)
     span.start = dt.replace(tzinfo=timezone.utc).timestamp()
+
+    # Since inferred span will later parent Lambda, preserve Lambda's current parent
+    span.parent_id = dd_trace_context.span_id
```

Comment:
This is important because we have the following code in `tracing.create_function_execution_span()`
```
if parent_span:
    span.parent_id = parent_span.span_id
```
where `parent_span` is the generated inferred span so the Lambda's root span's `parent_id` will be set to the inferred span's `span_id`

If there is an upstream Step Function and we saved its trace context in `dd_trace_context`, we want to preserve the parenting relationship and not let the inferred span completely erase it

This line solves the issue by making the inferred span be a child of the upstream service